File: solution/1248. Count Number of Nice Subarrays/1248_02.cpp
class Solution {
public:
    int numberOfSubarrays(vector<int>& nums, int k) {
        return atMost(nums, k) - atMost(nums, k - 1);
    }
    
    int atMost(vector<int>& nums, int k) {
        const int n = nums.size();
        int res = 0;
        for (int j = 0, i = 0; i < n; ++i) {
            k -= nums[i] & 1;
            while (k < 0)
                k += nums[j++] & 1;
            res += i - j + 1;
        }
        return res;
    }
};

File: solution/1248. Count Number of Nice Subarrays/1248_02.py
class Solution:
    def numberOfSubarrays(self, nums: List[int], k: int) -> int:
        def atMost(k):
            res = j = 0
            for i in range(len(nums)):
                k -= nums[i] % 2
                while k < 0:
                    k += nums[j] % 2
                    j += 1
                res += i - j + 1
            return res
        
        return atMost(k) - atMost(k - 1)
    
